added probe implementation to CompositeDiscoveryClient

sruffatti opened the pull request:
Fixes gh-1209

spencergibb reviewed (changes requested):
Please add a test

sruffatti commented:
Any additional feedback on this @spencergibb ?

sruffatti commented:
Any updates on this @spencergibb ?

sruffatti commented:
bump @spencergibb

sruffatti commented:
@OlgaMaciaszek Are you able to help me out here? This PR has been open for a while.

OlgaMaciaszek commented on spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/composite/CompositeDiscoveryClientUnitTests.java:
```diff
@@ -0,0 +1,105 @@
+     }
+ 
+     @Test
+     void getDescriptionShouldBeComposite() {
```
Please remove this test - we don't really need to test hard-coded `String` values.

OlgaMaciaszek commented on spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/composite/CompositeDiscoveryClientUnitTests.java:
```diff
@@ -0,0 +1,105 @@
+     }
+ 
+     @Test
+     void getInstancesByServiceIdShouldReturnServiceInstances() {
```
Please rename the test to `shouldRetrieveInstancesByServiceId`.

OlgaMaciaszek commented on spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/composite/CompositeDiscoveryClientUnitTests.java:
```diff
@@ -0,0 +1,105 @@
+     }
+ 
+     @Test
+     void getServicesShouldReturnServiceIds() {
```
Please rename the test to `shouldReturnServiceIds`.

OlgaMaciaszek commented on spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/composite/CompositeDiscoveryClientUnitTests.java:
```diff
@@ -0,0 +1,105 @@
+     }
+ 
+     @Test
+     void getDiscoveryClientsShouldReturnAllDiscoveryClients() {
```
Please rename the test to `shouldReturnAllDiscoveryClients`.

OlgaMaciaszek commented on spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/composite/CompositeDiscoveryClientUnitTests.java:
```diff
@@ -0,0 +1,105 @@
+     }
+ 
+     @Test
+     void probeShouldCallProbeInAllClientBeans() {
```
Please rename to `shouldCallProbeOnAllDiscoveryClients`.

OlgaMaciaszek commented on spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/composite/CompositeDiscoveryClientUnitTests.java:
```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
```
Please update the copyright dates to `2012-2023`.

OlgaMaciaszek commented on spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/composite/CompositeDiscoveryClient.java:
```diff
@@ -73,6 +73,15 @@ public List<String> getServices() {
 		return new ArrayList<>(services);
 	}
 
+	@Override
```
Please add your name with `@author` tag in the javadoc of the edited file. Please update the copyright date line.

OlgaMaciaszek reviewed (changes requested):
Hi @sruffatti, thanks for submitting the PR. Makes sense. There are some minor style issues. I have added comments - please address them. Also, could you please file this against `3.1.x` instead of `main`? It'll be considered a bugfix, so we'd like for it to also be included in the `2021.0.x` release train.

sruffatti commented:
> Hi @sruffatti, thanks for submitting the PR. Makes sense. There are some minor style issues. I have added comments - please address them. Also, could you please file this against `3.1.x` instead of `main`? It'll be considered a bugfix, so we'd like for it to also be included in the `2021.0.x` release train.

@OlgaMaciaszek I addressed your comments. Are you asking me to change the target branch to `3.1.x` instead `main`? When I do that I see more commits. Are you asking me to branch off `3.1.x`, add my changes, and then open a PR?

OlgaMaciaszek commented:
Hi @sruffatti,
> @OlgaMaciaszek I addressed your comments. Are you asking me to change the target branch to 3.1.x instead main? When I do that I see many commits included in the PR. Are you asking me to branch off 3.1.x, add my changes, and then open a PR? 

The second approach will probably be faster and easier, so it's a good idea.

sruffatti commented:
@OlgaMaciaszek I created another PR merging to 3.1.x here https://github.com/spring-cloud/spring-cloud-commons/pull/1239

OlgaMaciaszek commented:
Closing in favour of https://github.com/spring-cloud/spring-cloud-commons/pull/1239.
